### PR TITLE
**RFC:** Ensure cors allows cookies through

### PR DESCRIPTION
**Pull request by jspc**
Note: this PR also includes changes to help in development and a bug fix to `POST /upload`

This PR is in RFC, do not merge.

XHR doesn't like our CORS headers, thus not setting cookies. This means authentication bombs. This PR contains a multitude of fixes for this issue.

### CODE REVIEW

**Comment by jspc** [comment]
@jonathan-best: this branch should now be stable enough for you to use, should you want to.

File: api.go [hunk 1]
```diff
@@ -19,14 +21,27 @@ type ResponseBody struct {
     Message string
 }
 
+const (
```
Comment (jemrayfieldft) [inline api.go]: Can you make this env rather than hard coded.

Comment (jspc) [inline api.go]: both age and name? Or just one?

I don't see the value in doing that. In development the cookie persists as long as the app runs, and the app would have to restart for any change anyway so there is no danger of stale cookies.

Comment (jemrayfieldft) [inline api.go]: ok you know best.

**Comment by jemrayfieldft** [comment]
Hi,

Can you tell me how I might build the docker container


docker build .
Sending build context to Docker daemon 89.09 kB
Step 1 : FROM alpine
 ---> 4e38e38c8ce0
Step 2 : RUN apk --update add ca-certificates
 ---> Running in 22faa9eee32d
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
(1/1) Installing ca-certificates (20160104-r4)
Executing busybox-1.24.2-r9.trigger
Executing ca-certificates-20160104-r4.trigger
OK: 5 MiB in 12 packages
 ---> 0471abc40ea3
Removing intermediate container 22faa9eee32d
Step 3 : COPY ./s3-club-7 /s3-club-7
lstat s3-club-7: no such file or directory

Thanks
Jem

**Comment by jspc** [comment]
`make dist` will do it- that'll also make sure it compiles a linux binary,
even on a mac. The docs for this are proper shit (non existent?)- I'll open
a PR this afternoon.

**Comment by jemrayfieldft** [comment]
$ make dist
rm -v s3-club-7
rm: s3-club-7: No such file or directory
make: [clean] Error 1 (ignored)
CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -x
WORK=/var/folders/tg/l_lqwwjd4mn_5ncfhb6zxdzd9yd_fz/T/go-build886059175
aws.go:7:5: cannot find package "github.com/aws/aws-sdk-go/aws" in any of:
	/usr/local/Cellar/go/1.6/libexec/src/github.com/aws/aws-sdk-go/aws (from $GOROOT)
	/Users/jem.rayfield/dev/golang/src/github.com/aws/aws-sdk-go/aws (from $GOPATH)
aws.go:8:5: cannot find package "github.com/aws/aws-sdk-go/aws/session" in any of:
	/usr/local/Cellar/go/1.6/libexec/src/github.com/aws/aws-sdk-go/aws/session (from $GOROOT)
	/Users/jem.rayfield/dev/golang/src/github.com/aws/aws-sdk-go/aws/session (from $GOPATH)
aws.go:9:5: cannot find package "github.com/aws/aws-sdk-go/service/s3" in any of:
	/usr/local/Cellar/go/1.6/libexec/src/github.com/aws/aws-sdk-go/service/s3 (from $GOROOT)
	/Users/jem.rayfield/dev/golang/src/github.com/aws/aws-sdk-go/service/s3 (from $GOPATH)
main.go:8:5: cannot find package "github.com/gorilla/securecookie" in any of:
	/usr/local/Cellar/go/1.6/libexec/src/github.com/gorilla/securecookie (from $GOROOT)
	/Users/jem.rayfield/dev/golang/src/github.com/gorilla/securecookie (from $GOPATH)
make: *** [main] Error 1

**Comment by jemrayfieldft** [comment]
How do you set cluster and flex-api when using the docker image?

Cheers
Jem

**Pull request merged**